core/services/chainlink/config_general.go

```diff
@@ -136,10 +146,60 @@ func (g *generalConfig) TerraConfigs() terra.TerraConfigs {
 }
 
 func (g *generalConfig) Validate() error {
-	_, err := utils.MultiErrorList(multierr.Combine(g.c.Validate(), g.secrets.Validate()))
+	_, err := utils.MultiErrorList(multierr.Combine(
+		validateEnv(),
+		g.c.Validate(),
+		g.secrets.Validate()))
 	return err
 }
 
+//go:embed cfgtest/dump/empty-strings.env
+var emptyStringsEnv string
+
+var legacyEnvToV2 = map[string]string{
+	"CHAINLINK_DEV": "CL_DEV",
+
+	"DATABASE_URL":                            "CL_DATABASE_URL",
+	"DATABASE_BACKUP_URL":                     "CL_DATABASE_BACKUP_URL",
+	"SKIP_DATABASE_PASSWORD_COMPLEXITY_CHECK": "CL_DATABASE_ALLOW_SIMPLE_PASSWORDS",
+
+	"EXPLORER_ACCESS_KEY": "CL_EXPLORER_ACCESS_KEY",
+	"EXPLORER_SECRET":     "CL_EXPLORER_SECRET",
+
+	"PYROSCOPE_AUTH_TOKEN": "CL_PYROSCOPE_AUTH_TOKEN",
```

Comment:
There are more `PYROSCOPE_` env vars, I believe 2 more. Why don't we have them here as well?

Reply:
Only the auth token is a 'secret', and only secrets have v2 env vars. The rest of the config is only settable via the toml.